app/Jobs/PingCheck.php

```diff
@@ -306,4 +306,25 @@ private function defer($data)
             $this->deferred->put($device->max_depth, collect([$device->hostname => $data]));
         }
     }
+
+    /**
+     * Updates the device's hostnames to the overwrite IP if it exists
+     *
+     *
+     * @param $devices
+     */
+    private function updateDeviceHostnames($devices)
+    {
+        $updated_devices = clone $devices;
+
+        foreach ($devices as $hostname => $device) {
+            $updated_hostname = $device["overwrite_ip"] ?: $device["hostname"];
+            if ($updated_hostname != $hostname) {
```

Comment:
yeah i think i could write `$updated_hostname = Device::pollerTarget($device)` instead, but I'd need a function to update the keys of $this->devices and $this->tiered still. I can do it in the handle function as well, just thought I'd split it off to its own function. I think a cleaner way to do this would be to replace `$query->get()->keyBy('hostname')` with something that just calls `Device::pollerTarget` on the device, but I'm a bit new to PHP/not sure how to do that.

Reply:
Ah wait, I think `keyBy` also takes a callback. I can just pass in Device::pollerTarget and should be able to remove the function altogether.

Reply:
should be resolved, barring any style fixes